### PR TITLE
Added use case for Cluster Health Checking.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/917

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

/cc @detiber @vincepri

### CODE REVIEW

File: docs/staging-use-cases.md [hunk 1]
```diff
@@ -131,6 +131,23 @@ Multitenancy Management
...
+  - Describe nodes and provide details if they are ready/healthy or not ready/healthy.
+  - List conditions for any nodes which are `NotReady`, list information about allocated resources.
+
+- As an operator, given I have created a Kubernetes-conformant cluster with ClusterAPI, I want to check the kube-apiserver status.
+
+- As an operator, given I have created a Kubernetes-conformant cluster with ClusterAPI, I want to check the Kubernetes core components status, like etcd, ingress controller, other add-on components etc.
+
+- As an operator, given I have created a Kubernetes-conformant cluster with ClusterAPI, I want to check unhealthy Pods statuses in configured namespace.
+  - Provide the details on any pods which are unhealthy in `kube-system` namespace. Filter the unhealthy pods for their status(`kubectl get pods --show-labels -n kube-system | grep -vE "Running|Completed"`)
+  - Describe any pods which are not `Completed|Running`, list the Events to provide hints on the failure.
```
Comment (neolit123) [inline docs/staging-use-cases.md]: please uppercase `pods` -> `Pods` in the new text. multiple instances above.

File: docs/staging-use-cases.md [hunk 2]
```diff
@@ -131,6 +131,23 @@ Multitenancy Management
...
+  - Look for Pods which are `0/1` or `3/4` (not all containers are running).
```
Comment (neolit123) [inline docs/staging-use-cases.md]: > Look for Pods which don't have all of their containers running.

File: docs/staging-use-cases.md [hunk 3]
```diff
@@ -131,6 +131,23 @@ Multitenancy Management
...
+
+Cluster Health Checking is a service to provide the health status of Kubernetes cluster and its components.
+
+- As an operator, given I have created a Kubernetes-conformant cluster with ClusterAPI, I want to check the Kubernetes cluster node status.
+  - Describe nodes and provide details if they are ready/healthy or not ready/healthy.
+  - List conditions for any nodes which are `NotReady`, list information about allocated resources.
+
+- As an operator, given I have created a Kubernetes-conformant cluster with ClusterAPI, I want to check the kube-apiserver status.
+
+- As an operator, given I have created a Kubernetes-conformant cluster with ClusterAPI, I want to check the Kubernetes core components status, like etcd, ingress controller, other add-on components etc.
```
Comment (detiber) [inline docs/staging-use-cases.md]: This should probably be split into two use cases, with etcd being a separate item than add-on components such as ingress controller, with add-on components being marked with 🔭

File: docs/staging-use-cases.md [hunk 4]
```diff
@@ -131,6 +131,23 @@ Multitenancy Management
...
+
+- As an operator, given I have created a Kubernetes-conformant cluster with ClusterAPI, I want to check unhealthy Pods statuses in configured namespace.
```
Comment (detiber) [inline docs/staging-use-cases.md]: This should likely be marked with 🔭, this seems like something that would be out of scope for cluster-api proper, but should be able to be built with higher-level tooling on top of cluster-api.

**Comment by vincepri** [comment]
@gyliu513 Would you be able to address the comments?

**Comment by gyliu513** [comment]
@vincepri thanks for the reminder, will update soon.

File: docs/staging-use-cases.md (hunk 3 above)
Comment (gyliu513) [reply docs/staging-use-cases.md]: Done

File: docs/staging-use-cases.md (hunk 4 above)
Comment (gyliu513) [reply docs/staging-use-cases.md]: Done

File: docs/staging-use-cases.md (hunk 1 above)
Comment (gyliu513) [reply docs/staging-use-cases.md]: Done

File: docs/staging-use-cases.md (hunk 2 above)
Comment (gyliu513) [reply docs/staging-use-cases.md]: Done

**Comment by gyliu513** [comment]
@detiber @vincepri All comment are addressed now, thanks !

**Pull request merged**